roles/tas_single_node/tasks/certificates.yml

```diff
@@ -653,30 +653,262 @@
+      register: existing_rekor_keys
+
+    - name: Read existing rekor private key files
+      ansible.builtin.slurp:
+        src: "{{ item.path }}"
+      loop: "{{ existing_rekor_keys.files }}"
+      register: existing_keys
+
+- name: Set rekor_private_keys_changed
+  ansible.builtin.set_fact:
+    rekor_private_keys_changed: >-
+      {{ (existing_keys.results | length != (tas_single_node_rekor.private_keys | map(attribute='key') | list | length )
+      or (existing_keys.results | map(attribute='content') | map('b64decode') | list) != (tas_single_node_rekor.private_keys | map(attribute='key') | list ))
+      and (tas_single_node_rekor.private_keys | length > 0
+        or (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_rekor_use_provided_root_marker_file) | list | length != 0)) }}
+
+- name: Back-up changed rekor private keys
+  when: existing_keys.results | length > 0 and rekor_private_keys_changed
+  block:
+    - name: Identify private keys that need back-up
+      ansible.builtin.set_fact:
+        keys_to_backup: >-
+          {{ existing_keys.results |
+            selectattr('content', 'defined') |
+            map(attribute='content') |
+            map('b64decode') |
+            zip(existing_keys.results | map(attribute='source')) |
+            rejectattr(0, 'in', (tas_single_node_rekor.private_keys | map(attribute='key') | list )) |
+            map(attribute=1) |
+            list }}
+
+    - name: Back-up old rekor private keys
+      ansible.builtin.copy:
+        src: "{{ item }}"
+        dest: "{{ item + '.' + ansible_date_time.iso8601_basic }}"
+        remote_src: true
+        mode: '0600'
+      loop: "{{ keys_to_backup }}"
+
+    - name: Clean-up old rekor private keys
+      ansible.builtin.file:
+        path: "{{ item }}"
+        state: absent
+      loop: "{{ keys_to_backup }}"
+
+# Back-up rekor public keys
+- name: Get existing rekor public keys
+  block:
+    - name: Find existing rekor public key files
+      ansible.builtin.find:
+        paths: "{{ tas_single_node_certs_dir }}"
+        patterns: "rekor-pub-key*.pub"
+      register: existing_rekor_keys
+
+    - name: Read existing rekor public key files
+      ansible.builtin.slurp:
+        src: "{{ item.path }}"
+      loop: "{{ existing_rekor_keys.files }}"
+      register: existing_keys
+
+- name: Set rekor_public_keys_changed
+  ansible.builtin.set_fact:
+    rekor_public_keys_changed: >-
+      {{ (existing_keys.results | length != ( tas_single_node_rekor.public_keys | map(attribute='key') | list | length )
+      or (existing_keys.results | map(attribute='content') | map('b64decode') | list) != (tas_single_node_rekor.public_keys | map(attribute='key') | list ))
+      and (tas_single_node_rekor.public_keys | length  > 0
+        or (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_rekor_use_provided_root_marker_file ) | list | length != 0)) }}
+
+- name: Check changed keys
```

Comment:
I'm assuming this is to be removed before merging, just highlighting it here so as not to forget.

Reply:
ha ha yes will remove now

Reply:
:ballot_box_with_check: